core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -934,7 +934,15 @@ public GroupKey groupKey(ImmutableBitSet groupSet) {
    *
    * <p>This method of creating a group key does not allow you to group on new
    * expressions, only column projections, but is efficient, especially when you
-   * are coming from an existing {@link Aggregate}. */
+   * are coming from an existing {@link Aggregate}.
+   *
+   * <p>This method of creating a group key allow you to create groupKey be a
```

Comment:
How about "This method of creating a group key allows you to create a group key which is a superset of the union of the groupingSets, for example: group by 0, 1, 2, grouping sets((0, 1), 0). When converting RelNode To Sql, the {@link org.apache.calcite.rel.rel2sql.RelToSqlConverter#visit(Aggregate)} will generate the superset grouping set, then by having clause filtering it out"

Reply:
Yes. I've modified the javadoc with an example based on yours.

Reply:
done